src/Composer/Package/ExtraArchive.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace LastCall\DownloadsPlugin\Composer\Package;
+
+use Composer\Package\PackageInterface;
+use LastCall\DownloadsPlugin\Composer\Package\Cleaner\Cleaner;
+use LastCall\DownloadsPlugin\Composer\Package\Cleaner\CleanerInterface;
+use LastCall\DownloadsPlugin\Composer\Package\Tracking\ArchiveTracking;
+use LastCall\DownloadsPlugin\Enum\Type;
+use LastCall\DownloadsPlugin\Model\Hash;
+
+class ExtraArchive extends ExtraDownload implements ExtraArchiveInterface
```

Comment:
`ExtraArchive` > `ExtraDownload` > `Package`

May be 3 levels of inheritance this is not a good idea. But I am not sure how to implement using composition.